src/Psalm/Internal/Analyzer/Statements/Expression/Call/ArgumentsAnalyzer.php

```diff
@@ -639,9 +639,70 @@ public static function checkArgumentsMatch(
-            if ($arg->unpack && $function_param_count > $argument_offset) {
-                for ($i = $argument_offset; $i < $function_param_count; $i++) {
-                    $arg_function_params[$argument_offset][] = $function_params[$i];
+            if ($arg->unpack) {
+                if ($function_param_count > $argument_offset) {
+                    for ($i = $argument_offset; $i < $function_param_count; $i++) {
+                        $arg_function_params[$argument_offset][] = $function_params[$i];
+                    }
+                }
+
+                if (($arg_value_type = $statements_analyzer->node_data->getType($arg->value))
+                    && $arg_value_type->hasArray()) {
+                    /**
+                     * @psalm-suppress PossiblyUndefinedStringArrayOffset
+                     * @var TArray|TList|TKeyedArray
+                     */
+                    $array_type = $arg_value_type->getAtomicTypes()['array'];
+
+                    if ($array_type instanceof TKeyedArray) {
+                        $array_key_types = $array_type->getGenericArrayType()->getChildNodes()[0]->getChildNodes();
+
+                        foreach ($array_key_types as $array_key_type) {
+                            if (!$array_key_type instanceof Type\Atomic\TLiteralString
+                                || ($function_storage && !$function_storage->allow_named_arg_calls)) {
+                                continue;
+                            }
+
+                            $param_found = false;
+
+                            foreach ($function_params as $candidate_param) {
+                                if ($candidate_param->name === $array_key_type->value) {
+                                    if (isset($matched_args[$candidate_param->name])
+                                        && !$candidate_param->is_variadic) {
+                                        if (IssueBuffer::accepts(
+                                            new InvalidNamedArgument(
+                                                'Parameter $' . $array_key_type->value . ' has already been used in '
+                                                . ($cased_method_id ?: $method_id),
+                                                new CodeLocation($statements_analyzer, $arg),
+                                                (string)$method_id
+                                            ),
+                                            $statements_analyzer->getSuppressedIssues()
+                                        )) {
+                                            // fall through
+                                        }
+                                    }
+
+                                    $matched_args[$candidate_param->name] = true;
+                                    $param_found = true;
+                                    break;
+                                }
+                            }
```

Comment:
`$param_found` should be set to `true` if the function has a variadic argument.
Haven't tested it myself yet, but it looks like this would cause a false-positive: https://psalm.dev/r/cac7929f84

Reply:
You are right. I pushed the fix. Thanks!